### PR TITLE
Contracts: Implement rescue funds mechanism

### PR DESCRIPTION
**Pull request by lgalende**

### CODE REVIEW

File: contracts/ERC4626Adapter.sol [hunk 1]
```diff
@@ -94,6 +97,22 @@ contract ERC4626Adapter is IERC4626Adapter, ERC4626, Ownable {
...
+     * @dev Withdraw ERC20 tokens to an external account. To be used in order to withdraw claimed protocol rewards.
+     * @param token Address of the token to be withdrawn
+     * @param recipient Address where the tokens will be transferred to
+     * @param amount Amount of tokens to withdraw
+     */
+    function rescueFunds(address token, address recipient, uint256 amount) external override onlyOwner {
+        if (token == address(0)) revert TokenZero();
+        if (token == address(erc4626)) revert TokenERC4626();
+        if (recipient == address(0)) revert RecipientZero();
+        if (amount == 0) revert AmountZero();
```
Comment (facuspagnuolo) [inline contracts/ERC4626Adapter.sol:110]: Let's use the `ERC4626Adapter` prefix as we do in the rest of the Mimic contracts

File: test/ERC4626Adapter.test.ts [hunk 2]
```diff
@@ -150,6 +152,139 @@ describe('ERC4626 Adapter', () => {
...
+            )
+          })
+        })
+      })
+
+      context('when the token is the zero address', () => {
+        const tokenAddr = ZERO_ADDRESS
+
+        it('reverts', async () => {
+          await expect(erc4626Adapter.rescueFunds(tokenAddr, recipient.address, amount)).to.be.revertedWith('TokenZero')
```
Comment (facuspagnuolo) [inline test/ERC4626Adapter.test.ts:270]:
```suggestion
        const token = ZERO_ADDRESS

        it('reverts', async () => {
          await expect(erc4626Adapter.rescueFunds(token, recipient.address, amount)).to.be.revertedWith('TokenZero')
```

File: test/ERC4626Adapter.test.ts [hunk 3]
```diff
@@ -150,6 +152,139 @@ describe('ERC4626 Adapter', () => {
...
+              })
+
+              context('when withdrawing native tokens', () => {
+                const token = NATIVE_TOKEN_ADDRESS
+
+                it('reverts', async () => {
+                  await expect(erc4626Adapter.rescueFunds(token, recipient.address, amount)).to.be.revertedWith(
+                    'Address: call to non-contract'
+                  )
+                })
```
Comment (facuspagnuolo) [inline test/ERC4626Adapter.test.ts:225]: Any particular reason why not allowing to withdraw native tokens?

**Review by facuspagnuolo — APPROVED** [review]
Looks good

File: test/ERC4626Adapter.test.ts (hunk 3 above)
Comment (lgalende) [reply test/ERC4626Adapter.test.ts:225]: No reason why. I will add it

**Pull request merged**